packages/api/endpoints/granules.js

```diff
@@ -142,14 +141,10 @@ async function del(req, res) {
   }
 
   // remove files from s3
-  if (granule.files) {
-    await Promise.all(granule.files.map(async (file) => {
-      if (await fileExists(file.bucket, file.key)) {
-        return deleteS3Object(file.bucket, file.key);
-      }
-      return {};
-    }));
-  }
+  await pMap(
+    lodashGet(granule, 'files', []),
+    ({ bucket, key }) => deleteS3Object(bucket, key)
```

Comment:
Two notes here:

1. If you try to delete an object that does not exist no error is thrown, so there is no need to do an explicit "does it exist" check first.
2. Because of the response caching issues that we've seen with S3, this extra "does it exist" check can actually cause later calls to think that the object does exist even after it's been deleted. Just deleting it without checking will help cut down on the number of incorrect responses we get from S3.